packages/sfui/frameworks/vue/composables/usePagination/usePagination.ts

```diff
@@ -0,0 +1,28 @@
+import { ref, computed } from 'vue';
+import paginate from 'jw-paginate';
+
+export function usePagination(totalItems: number, currentPage = 1, pageSize = 10, maxPages = 1) {
+  const activePage = ref(currentPage);
+
+  const pagination = computed(() => {
+    return paginate(totalItems, activePage.value, pageSize, maxPages);
+  });
+
+  const next = () => (activePage.value += 1);
+
+  const prev = () => (activePage.value -= 1);
+
+  const onPageChange = (newPage: number) => (activePage.value = newPage);
+
+  return {
+    totalPages: computed(() => pagination.value.totalPages),
+    pages: computed(() => pagination.value.pages),
+    selectedPage: computed(() => pagination.value.currentPage),
+    endPage: computed(() => pagination.value.endPage),
+    startPage: computed(() => pagination.value.startPage),
+    maxVisiblePages: maxPages,
+    next,
+    prev,
+    onPageChange,
```

Comment:
those variables are not reused anywhere else, but thats cosmetic thing 
```suggestion
    next: () => (activePage.value += 1),
    prev: () => (activePage.value -= 1),
    onPageChange: (newPage: number) => (activePage.value = newPage),
```

Reply:
https://github.com/vuestorefront/storefront-ui/pull/2535/commits/3ac7e43a8ab6b69cd07bd650bc5f0163b76f477b